specification/security/resource-manager/Microsoft.Security/preview/2023-05-01-preview/customRecommedations.json

```diff
@@ -0,0 +1,384 @@
+            "schema": {
+              "$ref": "../../../common/v2/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "delete": {
+        "x-ms-examples": {
+          "Delete a custom recommendation over subscription scope": {
+            "$ref": "./examples/CustomRecommendations/DeleteBySubscriptionCustomRecommendation_example.json"
+          },
+          "Delete a custom recommendation over security connector scope": {
+            "$ref": "./examples/CustomRecommendations/DeleteBySecurityConnectorCustomRecommendation_example.json"
+          },
+          "Delete a custom recommendation over management group scope": {
+            "$ref": "./examples/CustomRecommendations/DeleteByManagementGroupCustomRecommendation_example.json"
+          }
+        },
+        "tags": [
+          "CustomRecommendations"
+        ],
+        "description": "Delete a custom recommendation over a given scope",
+        "operationId": "CustomRecommendations_Delete",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "#/parameters/Scope"
+          },
+          {
+            "$ref": "#/parameters/customRecommendationName"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK - custom recommendation was deleted"
+          },
+          "204": {
+            "description": "No Content - custom recommendation does not exist"
+          },
+          "default": {
+            "description": "Error response that describes why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/v2/types.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "CustomRecommendationsList": {
+      "type": "object",
+      "description": "A list of Custom Recommendations",
+      "properties": {
+        "value": {
+          "description": "Collection of Custom Recommendations",
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CustomRecommendation"
+          }
+        },
+        "nextLink": {
+          "description": "The link used to get the next page of operations.",
+          "type": "string"
+        }
+      }
+    },
+    "CustomRecommendation": {
+      "type": "object",
+      "description": "Custom Recommendation",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "describes Custom Recommendation properties.",
+          "$ref": "#/definitions/CustomRecommendationProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../common/v1/types.json#/definitions/Resource"
```

Comment:
>"../../../common/v1/types.json#/definitions/Resource" [](http://example.com/codeflow?start=18&length=53)

consider using "../../../../../common-types/resource-management/v5/types.json#/definitions/TrackedResource" which include "systemData":[https://github.com/Azure/azure-openapi-validator/blob/main/docs/required-read-only-system-data.md](https://github.com/Azure/azure-openapi-validator/blob/main/docs/required-read-only-system-data.md)

Reply:
It's not a tracked resource

Reply:
https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-contracts.md#system-metadata-for-all-azure-resources this rule requires all Azure resource, not only tracked resource